File: apps/admin/src/services/retention-policies.ts
import { apiClient } from "@silveryarn/web-shared/api";
import type { RetentionPolicy } from "@/types";

/** GET /retention-policies — decisions.md #56(Q5) admin 전용 보유기간 설정 목록. */
export async function listRetentionPolicies(): Promise<RetentionPolicy[]> {
  return apiClient.get<RetentionPolicy[]>("/retention-policies");
}

/** PUT /retention-policies/{category} — 보유일수 조정. 법무 결정 사항이 아니라
 * 운영 판단이라 하드코딩하지 않고 admin이 직접 바꾼다(마이그레이션 0012 주석 참조). */
export async function updateRetentionPolicy(
  category: string,
  retentionDays: number,
): Promise<RetentionPolicy> {
  return apiClient.put<RetentionPolicy>(`/retention-policies/${category}`, { retentionDays });
}

File: services/backend/src/core_service/modules/retention/api/v1/retention_policies.py
"""retention_policies 리소스 라우터 — decisions.md #56(Q5), apps/admin "보유기간
설정" 화면용. 보유기간 자체가 법무·정책 판단이라 admin 전용."""

import uuid
from datetime import datetime

from fastapi import APIRouter, Depends
from pydantic import BaseModel

from core_service.auth_deps import require_roles
from core_service.modules.retention.application.retention_policy_service import (
    RetentionPolicyService,
)
from core_service.modules.retention.deps import get_retention_policy_service
from core_service.shared.domain_enums import FamilyRole
from core_service.shared.schemas import DataResponse

router = APIRouter(prefix="/retention-policies", tags=["retention-policies"])


class RetentionPolicyResponse(BaseModel):
    id: uuid.UUID
    category: str
    retention_days: int
    updated_at: datetime


class RetentionPolicyUpdateRequest(BaseModel):
    retention_days: int


def _to_response(policy) -> RetentionPolicyResponse:  # noqa: ANN001 — RetentionPolicy 도메인 dataclass
    return RetentionPolicyResponse(
        id=policy.id,
        category=policy.category,
        retention_days=policy.retention_days,
        updated_at=policy.updated_at,
    )


@router.get("", response_model=DataResponse[list[RetentionPolicyResponse]])
async def list_retention_policies(
    service: RetentionPolicyService = Depends(get_retention_policy_service),
    _admin=Depends(require_roles(FamilyRole.ADMIN)),  # noqa: ANN001
) -> DataResponse[list[RetentionPolicyResponse]]:
    policies = await service.list_policies()
    return DataResponse(data=[_to_response(p) for p in policies])


@router.put("/{category}", response_model=DataResponse[RetentionPolicyResponse])
async def update_retention_policy(
    category: str,
    body: RetentionPolicyUpdateRequest,
    service: RetentionPolicyService = Depends(get_retention_policy_service),
    _admin=Depends(require_roles(FamilyRole.ADMIN)),  # noqa: ANN001
) -> DataResponse[RetentionPolicyResponse]:
    policy = await service.set_retention_days(category, body.retention_days)
    return DataResponse(data=_to_response(policy))
